inode.c

```
@@ -126,51 +197,32 @@ static struct dentry *simplefs_lookup(struct inode *dir,
     struct buffer_head *bh = NULL, *bh2 = NULL;
     struct simplefs_file_ei_block *eblock = NULL;
     struct simplefs_dir_block *dblock = NULL;
-    struct simplefs_file *f = NULL;
-    int ei, bi, fi;
+    int ei, bi, fi, chk;
 
     /* Check filename length */
     if (dentry->d_name.len > SIMPLEFS_FILENAME_LEN)
         return ERR_PTR(-ENAMETOOLONG);
 
     /* Read the directory block on disk */
+
+    chk = __file_lookup(dir, dentry, &ei, &bi, &fi);
+    if (chk != 1) {
+        if (!chk)
+            return ERR_PTR(-chk);
+        goto search_end;
+    }
```

error handling is broken.

`ERR_PTR(-0)` returns NULL, which is a valid negative dentry lookup result. This silently converts "not found" to success with NULL inode. The logic is inverted and confusing.

Proposed fix:
```c
  chk = __file_lookup(dir, dentry, &ei, &bi, &fi);
  if (chk < 0)
      return ERR_PTR(chk);  // I/O error
  if (chk == 0)
      goto search_end;  // Not found, return NULL dentry
```

---

Fixed it and also used kernel error code for checking